lang/java/avro/src/main/java/org/apache/avro/file/DataFileWriter.java

```diff
@@ -195,25 +195,31 @@ public boolean isFlushOnEveryBlock() {
   }
 
   /** Open a writer appending to an existing file.
+   * <strong>This method closes <code>in</code>.</strong>
    * @param in reading the existing file.
    * @param out positioned at the end of the existing file.
    */
   public DataFileWriter<D> appendTo(SeekableInput in, OutputStream out)
     throws IOException {
     assertNotOpen();
-    DataFileReader<D> reader =
-      new DataFileReader<D>(in, new GenericDatumReader<D>());
-    this.schema = reader.getSchema();
-    this.sync = reader.getHeader().sync;
-    this.meta.putAll(reader.getHeader().meta);
-    byte[] codecBytes = this.meta.get(DataFileConstants.CODEC);
-    if (codecBytes != null) {
-      String strCodec = new String(codecBytes, "UTF-8");
-      this.codec = CodecFactory.fromString(strCodec).createInstance();
-    } else {
-      this.codec = CodecFactory.nullCodec().createInstance();
+    DataFileReader<D> reader = null;
+    try {
+      reader = new DataFileReader<D>(in, new GenericDatumReader<D>());
+      this.schema = reader.getSchema();
+      this.sync = reader.getHeader().sync;
+      this.meta.putAll(reader.getHeader().meta);
+      byte[] codecBytes = this.meta.get(DataFileConstants.CODEC);
+      if (codecBytes != null) {
+        String strCodec = new String(codecBytes, "UTF-8");
+        this.codec = CodecFactory.fromString(strCodec).createInstance();
+      } else {
+        this.codec = CodecFactory.nullCodec().createInstance();
+      }
+    } finally {
+      if (reader != null)
+        reader.close();
+      in.close();
```

Comment:
We can but it would be a backward incompatible change as this method is public. 
If we would like to follow the usual pattern for this case we would need another method which does not close it and deprecate this one then remove this one in the next major release.
Or is it enough marking this JIRA with "incompatible change" and change as you are suggesting?